### DISCUSSION TITLE
Custom member fields can only be edited when set to publically visible

### DISCUSSION THREAD

**Discussion by jaapmurre** [ExpressionEngine/ExpressionEngine#2858] [Feature Requests]
**Description of the problem**

When adding a new custom member field it is _only_ visible in member/edit_profile if the field has been been set to be visble in public profiles. Otherwise, it is not shown.

This is a problem for member fields you do not want to show but keep for internal processing. 

**How To Reproduce**

Create new member field

For: Show in profile?
       When enabled, this field will be shown in public profiles.

Set to Yes

Now the field can be edited by members in member/edit_profile

Set to No

Now the field is not visible in member/edit_profile

**Environment Details:**
 - Version: 5.2.3
 - PHP Version: 7.0

**Comment by robinsowell** [comment]
That's intended behavior.  It shows in the control panel, under the member's profile.  So it allows admins with access to store data that the member can't see or edit.

It sounds like what you'd like to do is not show it on the public profile page, but still allow the member to see and edit the value on the frontend- correct?

You could edit the profile template to not show the custom fields on the public profile page, while leaving them on the edit page.

**Comment by intoeetive** [comment]
I can think of cases where the field could be set to visible, but member should not be able to edit it.
And the opposite way, when member can edit the value, but that should not be available in public profiles.

We may consider adding yet another setting for field, something like "is field editable"
